app/views/sessions/purchase_cart.html.erb

```diff
@@ -7,7 +7,11 @@
     </tr>
     <% if session[:cart] %>
         <% session[:cart].each do |item_id, quantity| %>
-        <% item = Item.find_by(id: item_id) %>
+            <% item = Item.find_by(id: item_id) %>
+            <!-- Line below generates receipt for purchasing this item, if possible -->
+            <% if logged_in? %>
```

Comment:
@rynkwn be careful about actually performing **logic** in views. It really should be performed in the controller.

@BenDrews Transactions are sensitive. You wouldn't people being able to see what you purchase on Amazon, for example. So I would either make `receipts#show` only viewable when logged in, or give a receipt a UUID for ID -- that way, you have to guess the URL to view it, e.g, `/receipts/213847127`